Modules/Rat_Board.py

```diff
@@ -0,0 +1,197 @@
+"""
+Rat_Board.py - Rescue board
+
+Handles Tracking individual Rescues from creation to completion
+
+Copyright (c) 2018 The Fuel Rats Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+
+This module is built on top of the Pydle system.
+"""
+import logging
+from uuid import UUID
+
+from Modules.rat_rescue import Rescue
+from config import Logging
+
+LOG = logging.getLogger(f"{Logging.base_logger}.{__name__}")
+
+
+class RescueBoardException(BaseException):
+    """
+    Generic Rescue Board Exception
+    """
+
+    def __init__(self, *args: object) -> None:
+        LOG.error(f"Rescueboard Exception raised!\nargs = {args}")
+        super().__init__(*args)
+
+
+class RescueNotFoundException(RescueBoardException):
+    """
+    Exception raised when a Rescue was not found
+    """
+    pass
+
+
+class RescueNotChangedException(RescueBoardException):
+    """
+    Exception raised when there was no net change to a Rescue.
+    """
+    pass
+
+
+class IndexNotFreeError(RescueBoardException):
+    """ Exception raised when someone attempts to attach a Rescue to a board index in-use"""
+    pass
+
+
+class RatBoard(object):
```

Comment:
Can you provide me with a __contains__ method? Might also need a __eq__ on the rescue object for that.

Reply:
Sure thing. I presume `__contains__ is for `Rescue` objects?

Reply:
Implemented, see latest commits